pkgs/by-name/xw/xwayland-satellite/package.nix

```diff
@@ -1,62 +1,75 @@
-{ lib
-, rustPlatform
-, fetchFromGitHub
-, pkg-config
-, xwayland
-, xcb-util-cursor
-, libxcb
-, nix-update-script
-, makeWrapper
+{
+  lib,
+  fetchFromGitHub,
+  libxcb,
+  makeBinaryWrapper,
+  pkg-config,
+  rustPlatform,
+  unstableGitUpdater,
+  xcb-util-cursor,
+  xwayland,
+  withSystemd ? true,
 }:
 
-rustPlatform.buildRustPackage rec {
+rustPlatform.buildRustPackage {
   pname = "xwayland-satellite";
-  version = "0.4";
+  version = "0.4-unstable-2024-09-15";
 
   src = fetchFromGitHub {
     owner = "Supreeeme";
     repo = "xwayland-satellite";
-    rev = "v${version}";
-    hash = "sha256-dwF9nI54a6Fo9XU5s4qmvMXSgCid3YQVGxch00qEMvI=";
+    rev = "b962a0f33b503aa39c9cf6919f488b664e5b79b4";
+    hash = "sha256-OANPb73V/RQDqtpIcbzeJ93KuOHKFQv+1xXC44Ut7tY=";
   };
 
-  cargoHash = "sha256-nKPSkHbh73xKWNpN/OpDmLnVmA3uygs3a+ejOhwU3yA=";
+  postPatch = ''
+    substituteInPlace resources/xwayland-satellite.service \
+      --replace-fail '/usr/local/bin' "$out/bin"
+  '';
+
+  cargoHash = "sha256-1EtwGMoLfYK0VZj8jdQiweO/RHGBzyEoeMEI4pmqfu8=";
 
   nativeBuildInputs = [
+    makeBinaryWrapper
     pkg-config
     rustPlatform.bindgenHook
-    makeWrapper
   ];
 
   buildInputs = [
     libxcb
     xcb-util-cursor
   ];
 
-  # disable Xwayland integration tests which need a running display server
-  checkFlags = [
-    "--exact"
-    "--skip=copy_from_wayland"
-    "--skip=copy_from_x11"
-    "--skip=input_focus"
-    "--skip=quick_delete"
-    "--skip=reparent"
-    "--skip=toplevel_flow"
-  ];
+  buildNoDefaultFeatures = true;
+  buildFeatures = lib.optional withSystemd "systemd";
+
+  # All integration tests require a running display server
+  doCheck = false;
+
+  postInstall = lib.optionalString withSystemd ''
+    install -Dm0644 resources/xwayland-satellite.service -t $out/lib/systemd/user
+  '';
 
-  postInstall = ''
+  postFixup = ''
     wrapProgram $out/bin/xwayland-satellite \
-      --prefix PATH : "${lib.makeBinPath [xwayland]}"
+      --prefix PATH : "${lib.makeBinPath [ xwayland ]}"
   '';
 
-  passthru.updateScript = nix-update-script { };
+  passthru.updateScript = unstableGitUpdater { tagPrefix = "v"; };
 
-  meta = with lib; {
-    description = "Rootless Xwayland integration to any Wayland compositor implementing xdg_wm_base";
+  meta = {
+    description = "Xwayland outside your Wayland";
```

Comment:
I know you didn't write this description, it's just copied from the GitHub, but is this... actually helpful? It's wrong, no? `xwayland-satellite` clearly runs within Wayland. and it's generally the wrong format for short `description`, right (of course, the previous description is also wrong/too long)?

If someone is scrolling the packages on `search.nixos.org` i don't actually see this description being very useful. should we maybe write our own?

Reply:
> It's wrong, no? xwayland-satellite clearly runs within Wayland.

Pretty sure it means outside your Wayland *compositor*. Regardless though, this is the official upstream tagline...I'm not sure it's our place to say it's "wrong"

>  and it's generally the wrong format for short description

What exactly is the "right" format?

> If someone is scrolling the packages on search.nixos.org i don't actually see this description being very useful.

I'd say it still gets the point across. And in the case someone wants to know more, that's why we have the `longDescription`

Reply:
> What exactly is the "right" format?

In general, a short description, and not a tagline. like you said, this is the "official upstream tagline".

Take [this package, for example](https://gitlab.com/adhami3310/Footage), which i discovered through not too long of just looking through nixpkgs
![Footage: Polish your videos](https://github.com/user-attachments/assets/8ab19dcf-9d60-49f7-8803-d97919993a99)
This tagline isn't very useful, so [nixpkgs has an actual description of the package](https://github.com/NixOS/nixpkgs/blob/5633bcff0c6162b9e4b5f1264264611e950c8ec7/pkgs/by-name/fo/footage/package.nix#L95) in that field

> Pretty sure it means outside your Wayland *compositor*

but this makes sense though. then this "tagline" becomes somewhat of a description. so i guess upstream is just missing a word or two from its description?
```suggestion
    description = "Xwayland outside your Wayland compositor";
```
i still think it ought to mention rootless Xwayland to "get the point across", but yeah this is what `longDescription` is for; i prefer adding this extra word because then at least the description doesn't feel outright like "what? that's wrong?"